src/directives/ng-file-select.ts

```diff
@@ -25,8 +25,23 @@ export class NgFileSelect {
 
   onFiles(): void {
     let files = this.el.nativeElement.files;
-    if (files.length) {
+    let invalidExtensionFiles: Array<string> = [];
+
+    if (files.length && this.uploader.allowedExtensions.length) {
+      for (var i = 0; i < files.length; i++) {
+        (this.uploader.allowedExtensions.indexOf(files[i].type) === -1) ? invalidExtensionFiles.push(files[i].name) : null;
```

Comment:
Please use long version of `if` here. Otherwise this looks good and after update I believe ready to merge.